<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Proteomics Data Viz | r-intro.utf8</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Proteomics Data Viz | r-intro.utf8" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Proteomics Data Viz | r-intro.utf8" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="plotting-with-ggplot2.html"/>
<link rel="next" href="summarizing-data.html"/>
<script src="libs/header-attrs-2.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction to R</a></li>
<li class="chapter" data-level="1" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html"><i class="fa fa-check"></i><b>1</b> Starting out in R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#variables"><i class="fa fa-check"></i><b>1.1</b> Variables</a></li>
<li class="chapter" data-level="1.2" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#saving-code-in-an-r-script"><i class="fa fa-check"></i><b>1.2</b> Saving code in an R script</a></li>
<li class="chapter" data-level="1.3" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#vectors"><i class="fa fa-check"></i><b>1.3</b> Vectors</a></li>
<li class="chapter" data-level="1.4" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#types-of-vector"><i class="fa fa-check"></i><b>1.4</b> Types of vector</a></li>
<li class="chapter" data-level="1.5" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#indexing-vectors"><i class="fa fa-check"></i><b>1.5</b> Indexing vectors</a></li>
<li class="chapter" data-level="1.6" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#sequences"><i class="fa fa-check"></i><b>1.6</b> Sequences</a></li>
<li class="chapter" data-level="1.7" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#functions"><i class="fa fa-check"></i><b>1.7</b> Functions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-frames.html"><a href="data-frames.html"><i class="fa fa-check"></i><b>2</b> Data frames</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-frames.html"><a href="data-frames.html#setting-up"><i class="fa fa-check"></i><b>2.1</b> Setting up</a></li>
<li class="chapter" data-level="2.2" data-path="data-frames.html"><a href="data-frames.html#loading-data"><i class="fa fa-check"></i><b>2.2</b> Loading data</a></li>
<li class="chapter" data-level="2.3" data-path="data-frames.html"><a href="data-frames.html#exploring"><i class="fa fa-check"></i><b>2.3</b> Exploring</a></li>
<li class="chapter" data-level="2.4" data-path="data-frames.html"><a href="data-frames.html#indexing-data-frames"><i class="fa fa-check"></i><b>2.4</b> Indexing data frames</a></li>
<li class="chapter" data-level="2.5" data-path="data-frames.html"><a href="data-frames.html#columns-are-vectors"><i class="fa fa-check"></i><b>2.5</b> Columns are vectors</a></li>
<li class="chapter" data-level="2.6" data-path="data-frames.html"><a href="data-frames.html#logical-indexing"><i class="fa fa-check"></i><b>2.6</b> Logical indexing</a></li>
<li class="chapter" data-level="2.7" data-path="data-frames.html"><a href="data-frames.html#factors"><i class="fa fa-check"></i><b>2.7</b> Factors</a></li>
<li class="chapter" data-level="2.8" data-path="data-frames.html"><a href="data-frames.html#readability-vs-tidyness"><i class="fa fa-check"></i><b>2.8</b> Readability vs tidyness</a></li>
<li class="chapter" data-level="2.9" data-path="data-frames.html"><a href="data-frames.html#sorting"><i class="fa fa-check"></i><b>2.9</b> Sorting</a></li>
<li class="chapter" data-level="2.10" data-path="data-frames.html"><a href="data-frames.html#joining-data-frames"><i class="fa fa-check"></i><b>2.10</b> Joining data frames</a></li>
<li class="chapter" data-level="2.11" data-path="data-frames.html"><a href="data-frames.html#further-reading"><i class="fa fa-check"></i><b>2.11</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html"><i class="fa fa-check"></i><b>3</b> Plotting with ggplot2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#elements-of-a-ggplot"><i class="fa fa-check"></i><b>3.1</b> Elements of a ggplot</a></li>
<li class="chapter" data-level="3.2" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#further-geoms"><i class="fa fa-check"></i><b>3.2</b> Further geoms</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#highlighting-subsets"><i class="fa fa-check"></i><b>3.3</b> Highlighting subsets</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#fine-tuning-a-plot"><i class="fa fa-check"></i><b>3.4</b> Fine-tuning a plot</a></li>
<li class="chapter" data-level="3.5" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#faceting"><i class="fa fa-check"></i><b>3.5</b> Faceting</a></li>
<li class="chapter" data-level="3.6" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#saving-ggplots"><i class="fa fa-check"></i><b>3.6</b> Saving ggplots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="proteomics-data-viz.html"><a href="proteomics-data-viz.html"><i class="fa fa-check"></i><b>4</b> Proteomics Data Viz</a>
<ul>
<li class="chapter" data-level="4.1" data-path="proteomics-data-viz.html"><a href="proteomics-data-viz.html#visualizing-proteomics-data-with-ggplot2"><i class="fa fa-check"></i><b>4.1</b> Visualizing Proteomics data with ggplot2</a></li>
<li class="chapter" data-level="4.2" data-path="proteomics-data-viz.html"><a href="proteomics-data-viz.html#exploring-the-data"><i class="fa fa-check"></i><b>4.2</b> Exploring the data</a></li>
<li class="chapter" data-level="4.3" data-path="proteomics-data-viz.html"><a href="proteomics-data-viz.html#plotting-interactions-types"><i class="fa fa-check"></i><b>4.3</b> Plotting interactions types</a></li>
<li class="chapter" data-level="4.4" data-path="proteomics-data-viz.html"><a href="proteomics-data-viz.html#individual-proteins"><i class="fa fa-check"></i><b>4.4</b> Individual Proteins</a></li>
<li class="chapter" data-level="" data-path="proteomics-data-viz.html"><a href="proteomics-data-viz.html#volcano-plot"><i class="fa fa-check"></i>Volcano Plot</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="summarizing-data.html"><a href="summarizing-data.html"><i class="fa fa-check"></i><b>5</b> Summarizing data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="summarizing-data.html"><a href="summarizing-data.html#summary-functions"><i class="fa fa-check"></i><b>5.1</b> Summary functions</a></li>
<li class="chapter" data-level="5.2" data-path="summarizing-data.html"><a href="summarizing-data.html#missing-values"><i class="fa fa-check"></i><b>5.2</b> Missing values</a></li>
<li class="chapter" data-level="5.3" data-path="summarizing-data.html"><a href="summarizing-data.html#grouped-summaries"><i class="fa fa-check"></i><b>5.3</b> Grouped summaries</a></li>
<li class="chapter" data-level="5.4" data-path="summarizing-data.html"><a href="summarizing-data.html#t-test"><i class="fa fa-check"></i><b>5.4</b> t-test</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="r-markdown.html"><a href="r-markdown.html"><i class="fa fa-check"></i><b>6</b> R Markdown</a>
<ul>
<li class="chapter" data-level="6.1" data-path="r-markdown.html"><a href="r-markdown.html#introduction-to-markdown"><i class="fa fa-check"></i><b>6.1</b> Introduction to markdown</a></li>
<li class="chapter" data-level="6.2" data-path="r-markdown.html"><a href="r-markdown.html#document-types"><i class="fa fa-check"></i><b>6.2</b> Document types</a></li>
<li class="chapter" data-level="6.3" data-path="r-markdown.html"><a href="r-markdown.html#vanilla-markdown"><i class="fa fa-check"></i><b>6.3</b> Vanilla Markdown</a></li>
<li class="chapter" data-level="6.4" data-path="r-markdown.html"><a href="r-markdown.html#r-markdown-1"><i class="fa fa-check"></i><b>6.4</b> R Markdown</a></li>
<li class="chapter" data-level="6.5" data-path="r-markdown.html"><a href="r-markdown.html#yaml-header"><i class="fa fa-check"></i><b>6.5</b> YAML header</a></li>
<li class="chapter" data-level="6.6" data-path="r-markdown.html"><a href="r-markdown.html#alternate-formats"><i class="fa fa-check"></i><b>6.6</b> Alternate Formats</a></li>
<li class="chapter" data-level="6.7" data-path="r-markdown.html"><a href="r-markdown.html#extras"><i class="fa fa-check"></i><b>6.7</b> Extras</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="next-steps.html"><a href="next-steps.html"><i class="fa fa-check"></i><b>7</b> Next steps</a>
<ul>
<li class="chapter" data-level="7.1" data-path="next-steps.html"><a href="next-steps.html#deepen-your-understanding"><i class="fa fa-check"></i><b>7.1</b> Deepen your understanding</a></li>
<li class="chapter" data-level="7.2" data-path="next-steps.html"><a href="next-steps.html#expand-your-vocabulary"><i class="fa fa-check"></i><b>7.2</b> Expand your vocabulary</a></li>
<li class="chapter" data-level="7.3" data-path="next-steps.html"><a href="next-steps.html#join-the-community"><i class="fa fa-check"></i><b>7.3</b> Join the community</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="proteomics-data-viz" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Proteomics Data Viz</h1>
<div id="visualizing-proteomics-data-with-ggplot2" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Visualizing Proteomics data with ggplot2</h2>
<p>Last session we worked on Cross-linking Mass Spectrometry Data. The data consisted of interaction between 300 yeast nuclear proteins.
We also learned about concepts behind <code>Grammer of Graphics</code> and plotting using <code>ggplot2</code>.</p>
<p>In this tutorial, we will be working on the data from Yeast Nuclear Protein interaction study using Cross-linking Mass Spectrometry.</p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="proteomics-data-viz.html#cb158-1"></a><span class="kw">library</span>(tidyverse)</span></code></pre></div>
<p>We continue with the Cross-linking Proteomics dataset from Cytoscape tutorial.
The data was in excel, therefore we first converted it into <code>Comma separated file (csv)</code> format.</p>
<div id="optional-reading-excel-file-into-r" class="section level3 extra" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Optional: Reading excel file into R</h3>
<p>If you want to load the excel files files directly to <code>R</code> then you can use another library <code>readxl</code>.</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="proteomics-data-viz.html#cb159-1"></a><span class="kw">library</span>(readxl)</span>
<span id="cb159-2"><a href="proteomics-data-viz.html#cb159-2"></a>nuclear_xl_ms_excel &lt;-<span class="st"> </span>readxl<span class="op">::</span><span class="kw">read_excel</span>(<span class="st">&quot;r-intro-files/Nuclear_XL_MS.xlsx&quot;</span>)</span>
<span id="cb159-3"><a href="proteomics-data-viz.html#cb159-3"></a><span class="kw">head</span>(nuclear_xl_ms_excel)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 8
##   Protein1 Protein2 NameProtein1 NameProtein2 PPINovelty PPIEvidenceInfo…
##   &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;        &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;           
## 1 P02293   P04911   H2B1         H2A1         Known      Structure       
## 2 P02293   P02309   H2B1         H4           Known      Structure       
## 3 P02994   P32471   EF1A         EF1B         Known      Structure       
## 4 P0CX51   P38011   RS16A        GBLP         Known      Structure       
## 5 P02406   P0CX49   RL28         RL18A        Novel      STRING          
## 6 P33297   P53549   PRS6A        PRS10        Known      Structure       
## # … with 2 more variables: TotalNumberOfCSMs &lt;dbl&gt;,
## #   NumberUniqueLysLysContacts &lt;dbl&gt;</code></pre>
</div>
</div>
<div id="exploring-the-data" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Exploring the data</h2>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="proteomics-data-viz.html#cb161-1"></a><span class="kw">library</span>(readxl)</span>
<span id="cb161-2"><a href="proteomics-data-viz.html#cb161-2"></a>nuclear_xl_ms &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="st">&quot;r-intro-files/Nuclear_XL_MS.csv&quot;</span>)</span>
<span id="cb161-3"><a href="proteomics-data-viz.html#cb161-3"></a><span class="kw">head</span>(nuclear_xl_ms)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 8
##   Protein1 Protein2 NameProtein1 NameProtein2 PPINovelty PPIEvidenceInfo…
##   &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;        &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;           
## 1 P02293   P04911   H2B1         H2A1         Known      Structure       
## 2 P02293   P02309   H2B1         H4           Known      Structure       
## 3 P02994   P32471   EF1A         EF1B         Known      Structure       
## 4 P0CX51   P38011   RS16A        GBLP         Known      Structure       
## 5 P02406   P0CX49   RL28         RL18A        Novel      STRING          
## 6 P33297   P53549   PRS6A        PRS10        Known      Structure       
## # … with 2 more variables: TotalNumberOfCSMs &lt;dbl&gt;,
## #   NumberUniqueLysLysContacts &lt;dbl&gt;</code></pre>
<p>Now lets examine the dataset with two base R functions <code>str</code> and <code>summary</code></p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="proteomics-data-viz.html#cb163-1"></a><span class="kw">str</span>(nuclear_xl_ms)</span></code></pre></div>
<pre><code>## Classes &#39;spec_tbl_df&#39;, &#39;tbl_df&#39;, &#39;tbl&#39; and &#39;data.frame&#39;: 228 obs. of  8 variables:
##  $ Protein1                  : chr  &quot;P02293&quot; &quot;P02293&quot; &quot;P02994&quot; &quot;P0CX51&quot; ...
##  $ Protein2                  : chr  &quot;P04911&quot; &quot;P02309&quot; &quot;P32471&quot; &quot;P38011&quot; ...
##  $ NameProtein1              : chr  &quot;H2B1&quot; &quot;H2B1&quot; &quot;EF1A&quot; &quot;RS16A&quot; ...
##  $ NameProtein2              : chr  &quot;H2A1&quot; &quot;H4&quot; &quot;EF1B&quot; &quot;GBLP&quot; ...
##  $ PPINovelty                : chr  &quot;Known&quot; &quot;Known&quot; &quot;Known&quot; &quot;Known&quot; ...
##  $ PPIEvidenceInfoGroup      : chr  &quot;Structure&quot; &quot;Structure&quot; &quot;Structure&quot; &quot;Structure&quot; ...
##  $ TotalNumberOfCSMs         : num  36 21 20 13 13 12 12 12 11 10 ...
##  $ NumberUniqueLysLysContacts: num  12 6 5 1 2 3 2 3 3 1 ...
##  - attr(*, &quot;spec&quot;)=
##   .. cols(
##   ..   Protein1 = col_character(),
##   ..   Protein2 = col_character(),
##   ..   NameProtein1 = col_character(),
##   ..   NameProtein2 = col_character(),
##   ..   PPINovelty = col_character(),
##   ..   PPIEvidenceInfoGroup = col_character(),
##   ..   TotalNumberOfCSMs = col_double(),
##   ..   NumberUniqueLysLysContacts = col_double()
##   .. )</code></pre>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="proteomics-data-viz.html#cb165-1"></a><span class="kw">summary</span>(nuclear_xl_ms)</span></code></pre></div>
<pre><code>##    Protein1           Protein2         NameProtein1       NameProtein2      
##  Length:228         Length:228         Length:228         Length:228        
##  Class :character   Class :character   Class :character   Class :character  
##  Mode  :character   Mode  :character   Mode  :character   Mode  :character  
##                                                                             
##                                                                             
##                                                                             
##   PPINovelty        PPIEvidenceInfoGroup TotalNumberOfCSMs
##  Length:228         Length:228           Min.   : 1.000   
##  Class :character   Class :character     1st Qu.: 1.000   
##  Mode  :character   Mode  :character     Median : 1.000   
##                                          Mean   : 2.706   
##                                          3rd Qu.: 3.000   
##                                          Max.   :36.000   
##  NumberUniqueLysLysContacts
##  Min.   : 1.000            
##  1st Qu.: 1.000            
##  Median : 1.000            
##  Mean   : 1.342            
##  3rd Qu.: 1.000            
##  Max.   :12.000</code></pre>
<p>While the data in <code>PPINovelty</code> and <code>PPIEvidenceInfoGroup</code> are characters/strings, they can also be thought of as categorical.</p>
<p>We will now look into the unique values for the PPINovelty and PPIEvidenceInfoGroup columns</p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="proteomics-data-viz.html#cb167-1"></a>nuclear_xl_ms<span class="op">$</span>PPINovelty <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unique</span>()</span></code></pre></div>
<pre><code>## [1] &quot;Known&quot; &quot;Novel&quot;</code></pre>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="proteomics-data-viz.html#cb169-1"></a>nuclear_xl_ms<span class="op">$</span>PPIEvidenceInfoGroup <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unique</span>()</span></code></pre></div>
<pre><code>## [1] &quot;Structure&quot;   &quot;STRING&quot;      &quot;APID&quot;        &quot;Unexplained&quot; &quot;Genetic&quot;</code></pre>
<p>R has a class for categorical data known as factors. We can convert these columns to factors and provide an order to those categories (levels). By default, R will order the levels of a factor alphabetically but we can override this behaviour by defining the level order.
Here we will order the Evidence based on Strength of evidence for the interaction</p>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="proteomics-data-viz.html#cb171-1"></a>nuclear_xl_ms &lt;-<span class="st"> </span>nuclear_xl_ms <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(</span>
<span id="cb171-2"><a href="proteomics-data-viz.html#cb171-2"></a>  <span class="dt">PPINovelty =</span> <span class="kw">factor</span>(PPINovelty),</span>
<span id="cb171-3"><a href="proteomics-data-viz.html#cb171-3"></a>  <span class="dt">PPIEvidenceInfoGroup =</span> <span class="kw">factor</span>(PPIEvidenceInfoGroup, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;Structure&quot;</span>,<span class="st">&quot;APID&quot;</span>, <span class="st">&quot;STRING&quot;</span>, <span class="st">&quot;Genetic&quot;</span>, <span class="st">&quot;Unexplained&quot;</span>))</span>
<span id="cb171-4"><a href="proteomics-data-viz.html#cb171-4"></a>)</span>
<span id="cb171-5"><a href="proteomics-data-viz.html#cb171-5"></a>nuclear_xl_ms <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str</span>()</span></code></pre></div>
<pre><code>## Classes &#39;spec_tbl_df&#39;, &#39;tbl_df&#39;, &#39;tbl&#39; and &#39;data.frame&#39;: 228 obs. of  8 variables:
##  $ Protein1                  : chr  &quot;P02293&quot; &quot;P02293&quot; &quot;P02994&quot; &quot;P0CX51&quot; ...
##  $ Protein2                  : chr  &quot;P04911&quot; &quot;P02309&quot; &quot;P32471&quot; &quot;P38011&quot; ...
##  $ NameProtein1              : chr  &quot;H2B1&quot; &quot;H2B1&quot; &quot;EF1A&quot; &quot;RS16A&quot; ...
##  $ NameProtein2              : chr  &quot;H2A1&quot; &quot;H4&quot; &quot;EF1B&quot; &quot;GBLP&quot; ...
##  $ PPINovelty                : Factor w/ 2 levels &quot;Known&quot;,&quot;Novel&quot;: 1 1 1 1 2 1 2 1 1 1 ...
##  $ PPIEvidenceInfoGroup      : Factor w/ 5 levels &quot;Structure&quot;,&quot;APID&quot;,..: 1 1 1 1 3 1 3 1 1 2 ...
##  $ TotalNumberOfCSMs         : num  36 21 20 13 13 12 12 12 11 10 ...
##  $ NumberUniqueLysLysContacts: num  12 6 5 1 2 3 2 3 3 1 ...</code></pre>
</div>
<div id="plotting-interactions-types" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Plotting interactions types</h2>
<p>Firstly we will plot the number of Known and novel interactions using with <code>geom_bar</code></p>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="proteomics-data-viz.html#cb173-1"></a><span class="kw">ggplot</span>(nuclear_xl_ms, <span class="kw">aes</span>(<span class="dt">x =</span> PPINovelty)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb173-2"><a href="proteomics-data-viz.html#cb173-2"></a><span class="kw">geom_bar</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb173-3"><a href="proteomics-data-viz.html#cb173-3"></a><span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Novelty&quot;</span>) <span class="op">+</span></span>
<span id="cb173-4"><a href="proteomics-data-viz.html#cb173-4"></a><span class="st">  </span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="Proteomics_data_viz_files/figure-html/unnamed-chunk-9-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>Now, we will rotate the bars to <strong>Y-Axis</strong> using <code>coord_flip()"</code></p>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="proteomics-data-viz.html#cb174-1"></a><span class="kw">ggplot</span>(nuclear_xl_ms, <span class="kw">aes</span>(<span class="dt">x =</span> PPINovelty)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb174-2"><a href="proteomics-data-viz.html#cb174-2"></a><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb174-3"><a href="proteomics-data-viz.html#cb174-3"></a><span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Novelty&quot;</span>) <span class="op">+</span></span>
<span id="cb174-4"><a href="proteomics-data-viz.html#cb174-4"></a><span class="st">        </span><span class="kw">coord_flip</span>() <span class="op">+</span></span>
<span id="cb174-5"><a href="proteomics-data-viz.html#cb174-5"></a><span class="st">  </span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="Proteomics_data_viz_files/figure-html/unnamed-chunk-10-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>Next, step would be to create a stacked bar chart by adding <strong>PPIEvidenceInfoGroup</strong> data on top of each bar</p>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="proteomics-data-viz.html#cb175-1"></a><span class="kw">ggplot</span>(nuclear_xl_ms, <span class="kw">aes</span>(PPINovelty)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb175-2"><a href="proteomics-data-viz.html#cb175-2"></a><span class="st">  </span><span class="kw">geom_bar</span>(<span class="kw">aes</span>(<span class="dt">fill=</span>PPIEvidenceInfoGroup)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb175-3"><a href="proteomics-data-viz.html#cb175-3"></a><span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Novelty&quot;</span>) <span class="op">+</span></span>
<span id="cb175-4"><a href="proteomics-data-viz.html#cb175-4"></a><span class="st">        </span><span class="kw">coord_flip</span>() <span class="op">+</span></span>
<span id="cb175-5"><a href="proteomics-data-viz.html#cb175-5"></a><span class="st">  </span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="Proteomics_data_viz_files/figure-html/unnamed-chunk-11-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="individual-proteins" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Individual Proteins</h2>
<p>First we will calculate how many times a protein appeared in <strong>NameProtein1</strong> column using <code>table</code> function and then sorting by descending order.
Next we will use <code>head()</code> function to print first five proteins with most observation.</p>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="proteomics-data-viz.html#cb176-1"></a><span class="kw">table</span>(nuclear_xl_ms<span class="op">$</span>NameProtein1) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>(.,<span class="dt">decreasing =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">head</span>()</span></code></pre></div>
<pre><code>## 
##  H2B1  EF1A RL27A    H3  EF3A  ODP2 
##    10     7     5     4     3     3</code></pre>
<p>We could do same thing for <strong>NameProtein2</strong> column as well.</p>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="proteomics-data-viz.html#cb178-1"></a><span class="kw">table</span>(nuclear_xl_ms<span class="op">$</span>NameProtein2)  <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sort</span>(.,<span class="dt">decreasing =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">head</span>()</span></code></pre></div>
<pre><code>## 
##    H3  RS15 NOP56  BFR1  PRS4 RL14A 
##     6     5     4     3     3     3</code></pre>
<p>Now we store the rows containing H2B1 and EF1A proteins in <strong>NameProtein1</strong> column in a datafame.</p>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="proteomics-data-viz.html#cb180-1"></a>two_protein_df&lt;-<span class="st"> </span>nuclear_xl_ms <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(NameProtein1 <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;H2B1&quot;</span>,<span class="st">&quot;EF1A&quot;</span>))</span>
<span id="cb180-2"><a href="proteomics-data-viz.html#cb180-2"></a></span>
<span id="cb180-3"><a href="proteomics-data-viz.html#cb180-3"></a><span class="kw">ggplot</span>(two_protein_df, <span class="kw">aes</span>(<span class="dt">x=</span>PPINovelty, <span class="dt">y=</span>NumberUniqueLysLysContacts)) <span class="op">+</span></span>
<span id="cb180-4"><a href="proteomics-data-viz.html#cb180-4"></a><span class="st">  </span><span class="kw">geom_col</span>(<span class="kw">aes</span>(<span class="dt">fill=</span>PPIEvidenceInfoGroup)) <span class="op">+</span></span>
<span id="cb180-5"><a href="proteomics-data-viz.html#cb180-5"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span> <span class="st">&quot;Novelty&quot;</span>,</span>
<span id="cb180-6"><a href="proteomics-data-viz.html#cb180-6"></a>  <span class="dt">y=</span> <span class="st">&quot;Number of Contacts&quot;</span>) <span class="op">+</span></span>
<span id="cb180-7"><a href="proteomics-data-viz.html#cb180-7"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>NameProtein1)<span class="op">+</span></span>
<span id="cb180-8"><a href="proteomics-data-viz.html#cb180-8"></a><span class="st">  </span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="Proteomics_data_viz_files/figure-html/unnamed-chunk-14-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano-plot" class="section level2 extra unnumbered" number="">
<h2>Volcano Plot</h2>
<p>In this section, we will see how to plot a volcano plot for a quantitative proteomics dataset.
This dataset is derived from label-free quantitative proteomics experiment investigating differences in protein profiles between Benign and
Malignant Prostate cancers.</p>
<p>The details can be found on <a href="https://bioinformatics.erc.monash.edu/apps/LFQ-Analyst/">LFQ-Analyst</a> under the <strong>Demo</strong> tab.</p>
<p>There are 20 samples in total with <code>n=10</code> in each group.</p>
<p>A moderated <em>t-test</em> was performed to find differentially expressed proteins in the dataset. Each row represents a protein along with log fold change and p-values.</p>
<p>In this tutorial we will visualise the results in the form of <strong>Volcano Plot</strong>.</p>
<p>Firstly, we will load the data.</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="proteomics-data-viz.html#cb181-1"></a>lfq_data&lt;-<span class="kw">read_csv</span>(<span class="st">&quot;r-intro-files/LFQ-Analyst_results.csv&quot;</span>)</span>
<span id="cb181-2"><a href="proteomics-data-viz.html#cb181-2"></a><span class="kw">nrow</span>(lfq_data)</span></code></pre></div>
<pre><code>## [1] 2389</code></pre>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="proteomics-data-viz.html#cb183-1"></a><span class="kw">ncol</span>(lfq_data)</span></code></pre></div>
<pre><code>## [1] 10</code></pre>
<p>The data has quantitative information about 2389 proteins and has 10 columns.</p>
<p>Now lets see the column names.</p>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="proteomics-data-viz.html#cb185-1"></a><span class="kw">colnames</span>(lfq_data)</span></code></pre></div>
<pre><code>##  [1] &quot;Gene Name&quot;                           
##  [2] &quot;Protein IDs&quot;                         
##  [3] &quot;Benign_vs_Malignant_log2 fold change&quot;
##  [4] &quot;Benign_vs_Malignant_p.val&quot;           
##  [5] &quot;Benign_vs_Malignant_p.adj&quot;           
##  [6] &quot;significant&quot;                         
##  [7] &quot;Benign_vs_Malignant_significant&quot;     
##  [8] &quot;imputed&quot;                             
##  [9] &quot;num_NAs&quot;                             
## [10] &quot;Protein.names&quot;</code></pre>
<p>For plotting the volcano plot, we need to focus on th three columns</p>
<ol style="list-style-type: decimal">
<li>Benign_vs_Malignant_log2 fold change</li>
<li>Benign_vs_Malignant_p.adj</li>
<li>significant</li>
</ol>
<p>Next we will on the fly convert FDR values to <code>-log10</code> and plot it against log2 fold change on the <strong>X-axis</strong></p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="proteomics-data-viz.html#cb187-1"></a>volcano_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(lfq_data, </span>
<span id="cb187-2"><a href="proteomics-data-viz.html#cb187-2"></a>                       <span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">`</span><span class="dt">Benign_vs_Malignant_log2 fold change</span><span class="st">`</span>, </span>
<span id="cb187-3"><a href="proteomics-data-viz.html#cb187-3"></a>                           <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(<span class="st">`</span><span class="dt">Benign_vs_Malignant_p.adj</span><span class="st">`</span>))) <span class="op">+</span><span class="st"> </span></span>
<span id="cb187-4"><a href="proteomics-data-viz.html#cb187-4"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">color=</span><span class="st">&quot;grey&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb187-5"><a href="proteomics-data-viz.html#cb187-5"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span> <span class="st">&quot;Log 2 fold change&quot;</span>,</span>
<span id="cb187-6"><a href="proteomics-data-viz.html#cb187-6"></a>       <span class="dt">y=</span> <span class="st">&quot;-log 10 FDR&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb187-7"><a href="proteomics-data-viz.html#cb187-7"></a><span class="st">  </span><span class="kw">theme_bw</span>()</span>
<span id="cb187-8"><a href="proteomics-data-viz.html#cb187-8"></a>volcano_plot</span></code></pre></div>
<p><img src="Proteomics_data_viz_files/figure-html/unnamed-chunk-17-1.png" width="576" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="proteomics-data-viz.html#cb188-1"></a>volcano_plot <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">data=</span> <span class="kw">filter</span>(lfq_data, significant<span class="op">==</span><span class="st">&quot;TRUE&quot;</span>), <span class="dt">color=</span><span class="st">&quot;black&quot;</span>)</span></code></pre></div>
<p><img src="Proteomics_data_viz_files/figure-html/unnamed-chunk-17-2.png" width="576" style="display: block; margin: auto;" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="plotting-with-ggplot2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="summarizing-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["r-intro.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
