<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Summarizing data | r-intro.utf8</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Summarizing data | r-intro.utf8" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Summarizing data | r-intro.utf8" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="proteomics-data-viz.html"/>
<link rel="next" href="r-markdown.html"/>
<script src="libs/header-attrs-2.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction to R</a></li>
<li class="chapter" data-level="1" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html"><i class="fa fa-check"></i><b>1</b> Starting out in R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#variables"><i class="fa fa-check"></i><b>1.1</b> Variables</a></li>
<li class="chapter" data-level="1.2" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#saving-code-in-an-r-script"><i class="fa fa-check"></i><b>1.2</b> Saving code in an R script</a></li>
<li class="chapter" data-level="1.3" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#vectors"><i class="fa fa-check"></i><b>1.3</b> Vectors</a></li>
<li class="chapter" data-level="1.4" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#types-of-vector"><i class="fa fa-check"></i><b>1.4</b> Types of vector</a></li>
<li class="chapter" data-level="1.5" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#indexing-vectors"><i class="fa fa-check"></i><b>1.5</b> Indexing vectors</a></li>
<li class="chapter" data-level="1.6" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#sequences"><i class="fa fa-check"></i><b>1.6</b> Sequences</a></li>
<li class="chapter" data-level="1.7" data-path="starting-out-in-r.html"><a href="starting-out-in-r.html#functions"><i class="fa fa-check"></i><b>1.7</b> Functions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-frames.html"><a href="data-frames.html"><i class="fa fa-check"></i><b>2</b> Data frames</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-frames.html"><a href="data-frames.html#setting-up"><i class="fa fa-check"></i><b>2.1</b> Setting up</a></li>
<li class="chapter" data-level="2.2" data-path="data-frames.html"><a href="data-frames.html#loading-data"><i class="fa fa-check"></i><b>2.2</b> Loading data</a></li>
<li class="chapter" data-level="2.3" data-path="data-frames.html"><a href="data-frames.html#exploring"><i class="fa fa-check"></i><b>2.3</b> Exploring</a></li>
<li class="chapter" data-level="2.4" data-path="data-frames.html"><a href="data-frames.html#indexing-data-frames"><i class="fa fa-check"></i><b>2.4</b> Indexing data frames</a></li>
<li class="chapter" data-level="2.5" data-path="data-frames.html"><a href="data-frames.html#columns-are-vectors"><i class="fa fa-check"></i><b>2.5</b> Columns are vectors</a></li>
<li class="chapter" data-level="2.6" data-path="data-frames.html"><a href="data-frames.html#logical-indexing"><i class="fa fa-check"></i><b>2.6</b> Logical indexing</a></li>
<li class="chapter" data-level="2.7" data-path="data-frames.html"><a href="data-frames.html#factors"><i class="fa fa-check"></i><b>2.7</b> Factors</a></li>
<li class="chapter" data-level="2.8" data-path="data-frames.html"><a href="data-frames.html#readability-vs-tidyness"><i class="fa fa-check"></i><b>2.8</b> Readability vs tidyness</a></li>
<li class="chapter" data-level="2.9" data-path="data-frames.html"><a href="data-frames.html#sorting"><i class="fa fa-check"></i><b>2.9</b> Sorting</a></li>
<li class="chapter" data-level="2.10" data-path="data-frames.html"><a href="data-frames.html#joining-data-frames"><i class="fa fa-check"></i><b>2.10</b> Joining data frames</a></li>
<li class="chapter" data-level="2.11" data-path="data-frames.html"><a href="data-frames.html#further-reading"><i class="fa fa-check"></i><b>2.11</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html"><i class="fa fa-check"></i><b>3</b> Plotting with ggplot2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#elements-of-a-ggplot"><i class="fa fa-check"></i><b>3.1</b> Elements of a ggplot</a></li>
<li class="chapter" data-level="3.2" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#further-geoms"><i class="fa fa-check"></i><b>3.2</b> Further geoms</a></li>
<li class="chapter" data-level="3.3" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#highlighting-subsets"><i class="fa fa-check"></i><b>3.3</b> Highlighting subsets</a></li>
<li class="chapter" data-level="3.4" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#fine-tuning-a-plot"><i class="fa fa-check"></i><b>3.4</b> Fine-tuning a plot</a></li>
<li class="chapter" data-level="3.5" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#faceting"><i class="fa fa-check"></i><b>3.5</b> Faceting</a></li>
<li class="chapter" data-level="3.6" data-path="plotting-with-ggplot2.html"><a href="plotting-with-ggplot2.html#saving-ggplots"><i class="fa fa-check"></i><b>3.6</b> Saving ggplots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="proteomics-data-viz.html"><a href="proteomics-data-viz.html"><i class="fa fa-check"></i><b>4</b> Proteomics Data Viz</a>
<ul>
<li class="chapter" data-level="4.1" data-path="proteomics-data-viz.html"><a href="proteomics-data-viz.html#visualizing-proteomics-data-with-ggplot2"><i class="fa fa-check"></i><b>4.1</b> Visualizing Proteomics data with ggplot2</a></li>
<li class="chapter" data-level="4.2" data-path="proteomics-data-viz.html"><a href="proteomics-data-viz.html#exploring-the-data"><i class="fa fa-check"></i><b>4.2</b> Exploring the data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="summarizing-data.html"><a href="summarizing-data.html"><i class="fa fa-check"></i><b>5</b> Summarizing data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="summarizing-data.html"><a href="summarizing-data.html#summary-functions"><i class="fa fa-check"></i><b>5.1</b> Summary functions</a></li>
<li class="chapter" data-level="5.2" data-path="summarizing-data.html"><a href="summarizing-data.html#missing-values"><i class="fa fa-check"></i><b>5.2</b> Missing values</a></li>
<li class="chapter" data-level="5.3" data-path="summarizing-data.html"><a href="summarizing-data.html#grouped-summaries"><i class="fa fa-check"></i><b>5.3</b> Grouped summaries</a></li>
<li class="chapter" data-level="5.4" data-path="summarizing-data.html"><a href="summarizing-data.html#t-test"><i class="fa fa-check"></i><b>5.4</b> t-test</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="r-markdown.html"><a href="r-markdown.html"><i class="fa fa-check"></i><b>6</b> R Markdown</a>
<ul>
<li class="chapter" data-level="6.1" data-path="r-markdown.html"><a href="r-markdown.html#introduction-to-markdown"><i class="fa fa-check"></i><b>6.1</b> Introduction to markdown</a></li>
<li class="chapter" data-level="6.2" data-path="r-markdown.html"><a href="r-markdown.html#document-types"><i class="fa fa-check"></i><b>6.2</b> Document types</a></li>
<li class="chapter" data-level="6.3" data-path="r-markdown.html"><a href="r-markdown.html#vanilla-markdown"><i class="fa fa-check"></i><b>6.3</b> Vanilla Markdown</a></li>
<li class="chapter" data-level="6.4" data-path="r-markdown.html"><a href="r-markdown.html#r-markdown-1"><i class="fa fa-check"></i><b>6.4</b> R Markdown</a></li>
<li class="chapter" data-level="6.5" data-path="r-markdown.html"><a href="r-markdown.html#yaml-header"><i class="fa fa-check"></i><b>6.5</b> YAML header</a></li>
<li class="chapter" data-level="6.6" data-path="r-markdown.html"><a href="r-markdown.html#alternate-formats"><i class="fa fa-check"></i><b>6.6</b> Alternate Formats</a></li>
<li class="chapter" data-level="6.7" data-path="r-markdown.html"><a href="r-markdown.html#extras"><i class="fa fa-check"></i><b>6.7</b> Extras</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="next-steps.html"><a href="next-steps.html"><i class="fa fa-check"></i><b>7</b> Next steps</a>
<ul>
<li class="chapter" data-level="7.1" data-path="next-steps.html"><a href="next-steps.html#deepen-your-understanding"><i class="fa fa-check"></i><b>7.1</b> Deepen your understanding</a></li>
<li class="chapter" data-level="7.2" data-path="next-steps.html"><a href="next-steps.html#expand-your-vocabulary"><i class="fa fa-check"></i><b>7.2</b> Expand your vocabulary</a></li>
<li class="chapter" data-level="7.3" data-path="next-steps.html"><a href="next-steps.html#join-the-community"><i class="fa fa-check"></i><b>7.3</b> Join the community</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="summarizing-data" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Summarizing data</h1>
<p>Having loaded and thoroughly explored a data set, we are ready to distill it down to concise conclusions. At its simplest, this involves calculating summary statistics like counts, means, and standard deviations. Beyond this is the fitting of models, and hypothesis testing and confidence interval calculation. R has a huge number of packages devoted to these tasks and this is a large part of its appeal, but is beyond the scope of today.</p>
<p>Loading the data as before, if you have not already done so:</p>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="summarizing-data.html#cb171-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb171-2"><a href="summarizing-data.html#cb171-2"></a></span>
<span id="cb171-3"><a href="summarizing-data.html#cb171-3"></a>geo &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="st">&quot;r-intro-files/geo.csv&quot;</span>)</span>
<span id="cb171-4"><a href="summarizing-data.html#cb171-4"></a>gap &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="st">&quot;r-intro-files/gap-minder.csv&quot;</span>)</span>
<span id="cb171-5"><a href="summarizing-data.html#cb171-5"></a>gap_geo &lt;-<span class="st"> </span><span class="kw">left_join</span>(gap, geo, <span class="dt">by=</span><span class="st">&quot;name&quot;</span>)</span></code></pre></div>
<div id="summary-functions" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Summary functions</h2>
<p>R has a variety of functions for summarizing a vector, including: <code>sum</code>, <code>mean</code>, <code>min</code>, <code>max</code>, <code>median</code>, <code>sd</code>.</p>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="summarizing-data.html#cb172-1"></a><span class="kw">mean</span>( <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>) )</span></code></pre></div>
<pre><code>## [1] 2.5</code></pre>
<p>We can use these on the Gapminder data.</p>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="summarizing-data.html#cb174-1"></a>gap2010 &lt;-<span class="st"> </span><span class="kw">filter</span>(gap_geo, year <span class="op">==</span><span class="st"> </span><span class="dv">2010</span>)</span>
<span id="cb174-2"><a href="summarizing-data.html#cb174-2"></a><span class="kw">sum</span>(gap2010<span class="op">$</span>population)</span></code></pre></div>
<pre><code>## [1] 6949495061</code></pre>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="summarizing-data.html#cb176-1"></a><span class="kw">mean</span>(gap2010<span class="op">$</span>life_exp)</span></code></pre></div>
<pre><code>## [1] NA</code></pre>
</div>
<div id="missing-values" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Missing values</h2>
<p>Why did <code>mean</code> fail? The reason is that <code>life_exp</code> contains missing values (<code>NA</code>).</p>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="summarizing-data.html#cb178-1"></a>gap2010<span class="op">$</span>life_exp</span></code></pre></div>
<pre><code>##   [1] 56.20 76.31 76.55 82.66 60.08 76.85 75.82 73.34 81.98 80.50 69.13 73.79
##  [13] 76.03 70.39 76.68 70.43 79.98 71.38 61.82 72.13 71.64 76.75 57.06 74.19
##  [25] 77.08 73.86 57.89 57.73 66.12 57.25 81.29 72.45 47.48 56.49 79.12 74.59
##  [37] 76.44 65.93 57.53 60.43 80.40 56.34 76.33 78.39 79.88 77.47 79.49 63.69
##  [49] 73.04 74.60 76.72 70.52 74.11 60.93 61.66 76.00 61.30 65.28 80.00 81.42
##  [61] 62.86 65.55 72.82 80.09 62.16 80.41 71.34 71.25 57.99 55.65 65.49 32.11
##  [73] 71.58 82.61 74.52 82.03 66.20 69.90 74.45 67.24 80.38 81.42 81.69 74.66
##  [85] 82.85 75.78 68.37 62.76 60.73 70.10 80.13 78.20 68.45 63.80 73.06 79.85
##  [97] 46.50 60.77 76.10    NA 73.17 81.35 74.01 60.84 53.07 74.46 77.91 59.46
## [109] 80.28 63.72 68.23 73.42 75.47 65.38 69.74    NA 66.18 76.36 73.55 54.48
## [121] 66.84 58.60    NA 68.26 80.73 80.90 77.36 58.78 60.53 81.04 76.09 65.33
## [133]    NA 77.85 58.70 74.07 77.92 69.03 76.30 79.84 79.52 73.66 69.24 64.59
## [145]    NA 75.48 71.64 71.46    NA 68.91 75.13 64.01 74.65 73.38 55.05 82.69
## [157] 75.52 79.45 61.71 53.13 54.27 81.94 74.42 66.29 70.32 46.98 81.52 82.21
## [169] 76.15 79.19 69.61 59.30 76.57 71.10 58.74 69.86 72.56 76.89 78.21 67.94
## [181]    NA 56.81 70.41 76.51 80.34 78.74 76.36 68.77 63.02 75.41 72.27 73.07
## [193] 67.51 52.02 49.57 58.13</code></pre>
<p>R will not ignore these unless we explicitly tell it to with <code>na.rm=TRUE</code>.</p>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="summarizing-data.html#cb180-1"></a><span class="kw">mean</span>(gap2010<span class="op">$</span>life_exp, <span class="dt">na.rm=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] 70.34005</code></pre>
<p>Ideally we should also use <code>weighted.mean</code> here, to take population into account.</p>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="summarizing-data.html#cb182-1"></a><span class="kw">weighted.mean</span>(gap2010<span class="op">$</span>life_exp, gap2010<span class="op">$</span>population, <span class="dt">na.rm=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] 70.96192</code></pre>
<p><code>NA</code> is a special value. If we try to calculate with <code>NA</code>, the result is <code>NA</code></p>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="summarizing-data.html#cb184-1"></a><span class="ot">NA</span> <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span></code></pre></div>
<pre><code>## [1] NA</code></pre>
<p><code>is.na</code> can be used to detect <code>NA</code> values, or <code>na.omit</code> can be used to directly remove rows of a data frame containing them.</p>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="summarizing-data.html#cb186-1"></a><span class="kw">is.na</span>( <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="ot">NA</span>,<span class="dv">3</span>) )</span></code></pre></div>
<pre><code>## [1] FALSE FALSE  TRUE FALSE</code></pre>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="summarizing-data.html#cb188-1"></a>cleaned &lt;-<span class="st"> </span><span class="kw">filter</span>(gap2010, <span class="op">!</span><span class="kw">is.na</span>(life_exp))</span>
<span id="cb188-2"><a href="summarizing-data.html#cb188-2"></a><span class="kw">weighted.mean</span>(cleaned<span class="op">$</span>life_exp, cleaned<span class="op">$</span>population)</span></code></pre></div>
<pre><code>## [1] 70.96192</code></pre>
</div>
<div id="grouped-summaries" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Grouped summaries</h2>
<p>The <code>summarize</code> function in <code>dplyr</code> allows summary functions to be applied to data frames.</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="summarizing-data.html#cb190-1"></a><span class="kw">summarize</span>(gap2010, <span class="dt">mean_life_exp=</span><span class="kw">weighted.mean</span>(life_exp, population, <span class="dt">na.rm=</span><span class="ot">TRUE</span>))</span></code></pre></div>
<pre><code>## # A tibble: 1 x 1
##   mean_life_exp
##           &lt;dbl&gt;
## 1          71.0</code></pre>
<p>So far unremarkable, but <code>summarize</code> comes into its own when the <code>group_by</code> “adjective” is used.</p>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="summarizing-data.html#cb192-1"></a><span class="kw">summarize</span>(</span>
<span id="cb192-2"><a href="summarizing-data.html#cb192-2"></a>    <span class="kw">group_by</span>(gap_geo, year), </span>
<span id="cb192-3"><a href="summarizing-data.html#cb192-3"></a>    <span class="dt">mean_life_exp=</span><span class="kw">weighted.mean</span>(life_exp, population, <span class="dt">na.rm=</span><span class="ot">TRUE</span>))</span></code></pre></div>
<pre><code>## # A tibble: 22 x 2
##     year mean_life_exp
##    &lt;dbl&gt;         &lt;dbl&gt;
##  1  1800          30.9
##  2  1810          31.1
##  3  1820          31.2
##  4  1830          31.4
##  5  1840          31.4
##  6  1850          31.6
##  7  1860          30.3
##  8  1870          31.5
##  9  1880          32.0
## 10  1890          32.5
## # … with 12 more rows</code></pre>
<div id="challenge-summarizing" class="section level3 unnumbered challenge" number="">
<h3>Challenge: summarizing</h3>
<p>What is the total population for each year? Plot the result.</p>
<p>Advanced: What is the total GDP for each year? For this you will first need to calculate GDP per capita times the population of each country.</p>
</div>
<div id="section-2" class="section level3 unnumbered" number="">
<h3 class="unnumbered" number=""></h3>
<p><code>group_by</code> can be used to group by multiple columns, much like <code>count</code>. We can use this to see how the rest of the world is catching up to OECD nations in terms of life expectancy.</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="summarizing-data.html#cb194-1"></a>result &lt;-<span class="st"> </span><span class="kw">summarize</span>(</span>
<span id="cb194-2"><a href="summarizing-data.html#cb194-2"></a>    <span class="kw">group_by</span>(gap_geo,year,oecd), </span>
<span id="cb194-3"><a href="summarizing-data.html#cb194-3"></a>    <span class="dt">mean_life_exp=</span><span class="kw">weighted.mean</span>(life_exp, population, <span class="dt">na.rm=</span><span class="ot">TRUE</span>))</span>
<span id="cb194-4"><a href="summarizing-data.html#cb194-4"></a>result</span></code></pre></div>
<pre><code>## # A tibble: 44 x 3
## # Groups:   year [22]
##     year oecd  mean_life_exp
##    &lt;dbl&gt; &lt;lgl&gt;         &lt;dbl&gt;
##  1  1800 FALSE          29.9
##  2  1800 TRUE           34.7
##  3  1810 FALSE          29.9
##  4  1810 TRUE           35.2
##  5  1820 FALSE          30.0
##  6  1820 TRUE           35.9
##  7  1830 FALSE          30.0
##  8  1830 TRUE           36.2
##  9  1840 FALSE          30.0
## 10  1840 TRUE           36.2
## # … with 34 more rows</code></pre>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="summarizing-data.html#cb196-1"></a><span class="kw">ggplot</span>(result, <span class="kw">aes</span>(<span class="dt">x=</span>year,<span class="dt">y=</span>mean_life_exp,<span class="dt">color=</span>oecd)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>()</span></code></pre></div>
<p><img src="summarizing_files/figure-html/unnamed-chunk-12-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>A similar plot could be produced using <code>geom_smooth</code>. Differences here are that we have full control over the summarization process so we were able to use the exact summarization method we want (<code>weighted.mean</code> for each year), and we have access to the resulting numeric data as well as the plot. We have reduced a large data set down to a smaller one that distills out one of the stories present in this data. However the earlier visualization and exploration activity using <code>ggplot2</code> was essential. It gave us an idea of what sort of variability was present in the data, and any unexpected issues the data might have.</p>
</div>
</div>
<div id="t-test" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> t-test</h2>
<p>We will finish this section by demonstrating a t-test. The main point of this section is to give a flavour of how statistical tests work in R, rather than the details of what a t-test does.</p>
<p>Has life expectancy increased from 2000 to 2010?</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="summarizing-data.html#cb197-1"></a>gap2000 &lt;-<span class="st"> </span><span class="kw">filter</span>(gap_geo, year <span class="op">==</span><span class="st"> </span><span class="dv">2000</span>)</span>
<span id="cb197-2"><a href="summarizing-data.html#cb197-2"></a>gap2010 &lt;-<span class="st"> </span><span class="kw">filter</span>(gap_geo, year <span class="op">==</span><span class="st"> </span><span class="dv">2010</span>)</span>
<span id="cb197-3"><a href="summarizing-data.html#cb197-3"></a></span>
<span id="cb197-4"><a href="summarizing-data.html#cb197-4"></a><span class="kw">t.test</span>(gap2010<span class="op">$</span>life_exp, gap2000<span class="op">$</span>life_exp)</span></code></pre></div>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  gap2010$life_exp and gap2000$life_exp
## t = 3.0341, df = 374.98, p-value = 0.002581
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  1.023455 4.792947
## sample estimates:
## mean of x mean of y 
##  70.34005  67.43185</code></pre>
<p>Statistical routines often have many ways to tweak the details of their operation. These are specified by further arguments to the function call, to override the default behaviour. By default, <code>t.test</code> performs an unpaired t-test, but these are repeated observations of the same countries. We can specify <code>paired=TRUE</code> to <code>t.test</code> to perform a paired sample t-test and gain some statistical power. Check this by looking at the help page with <code>?t.test</code>.</p>
<p>It’s important to first check that both data frames are in the same order.</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="summarizing-data.html#cb199-1"></a><span class="kw">all</span>(gap2000<span class="op">$</span>name <span class="op">==</span><span class="st"> </span>gap2010<span class="op">$</span>name)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="summarizing-data.html#cb201-1"></a><span class="kw">t.test</span>(gap2010<span class="op">$</span>life_exp, gap2000<span class="op">$</span>life_exp, <span class="dt">paired=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>## 
##  Paired t-test
## 
## data:  gap2010$life_exp and gap2000$life_exp
## t = 13.371, df = 188, p-value &lt; 2.2e-16
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  2.479153 3.337249
## sample estimates:
## mean of the differences 
##                2.908201</code></pre>
<p>When performing a statistical test, it’s good practice to visualize the data to make sure there is nothing funny going on.</p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="summarizing-data.html#cb203-1"></a><span class="kw">plot</span>(gap2000<span class="op">$</span>life_exp, gap2010<span class="op">$</span>life_exp)</span>
<span id="cb203-2"><a href="summarizing-data.html#cb203-2"></a><span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code></pre></div>
<p><img src="summarizing_files/figure-html/unnamed-chunk-15-1.png" width="336" style="display: block; margin: auto;" /></p>
<p>This is a visual confirmation of the t-test result. If there were no difference between the years then points would lie approximately evenly above and below the diagonal line, which is clearly not the case. However the outlier may warrant investigation.</p>
<div id="thinking-in-r" class="section level3 extra unnumbered" number="">
<h3>Thinking in R</h3>
<p>The result of a t-test is actually a value we can manipulate further. Two functions help us here. <code>class</code> gives the “public face” of a value, and <code>typeof</code> gives its underlying type, the way R thinks of it internally. For example numbers are “numeric” and have some representation in computer memory, either “integer” for whole numbers only, or “double” which can hold fractional numbers (stored in memory in a base-2 version of scientific notation).</p>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="summarizing-data.html#cb204-1"></a><span class="kw">class</span>(<span class="dv">42</span>)</span></code></pre></div>
<pre><code>## [1] &quot;numeric&quot;</code></pre>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="summarizing-data.html#cb206-1"></a><span class="kw">typeof</span>(<span class="dv">42</span>)</span></code></pre></div>
<pre><code>## [1] &quot;double&quot;</code></pre>
<p>Let’s look at the result of a t-test:</p>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="summarizing-data.html#cb208-1"></a>result &lt;-<span class="st"> </span><span class="kw">t.test</span>(gap2010<span class="op">$</span>life_exp, gap2000<span class="op">$</span>life_exp, <span class="dt">paired=</span><span class="ot">TRUE</span>)</span>
<span id="cb208-2"><a href="summarizing-data.html#cb208-2"></a></span>
<span id="cb208-3"><a href="summarizing-data.html#cb208-3"></a><span class="kw">class</span>(result)</span></code></pre></div>
<pre><code>## [1] &quot;htest&quot;</code></pre>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="summarizing-data.html#cb210-1"></a><span class="kw">typeof</span>(result)</span></code></pre></div>
<pre><code>## [1] &quot;list&quot;</code></pre>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="summarizing-data.html#cb212-1"></a><span class="kw">names</span>(result)</span></code></pre></div>
<pre><code>##  [1] &quot;statistic&quot;   &quot;parameter&quot;   &quot;p.value&quot;     &quot;conf.int&quot;    &quot;estimate&quot;   
##  [6] &quot;null.value&quot;  &quot;stderr&quot;      &quot;alternative&quot; &quot;method&quot;      &quot;data.name&quot;</code></pre>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="summarizing-data.html#cb214-1"></a>result<span class="op">$</span>p.value</span></code></pre></div>
<pre><code>## [1] 4.301261e-29</code></pre>
<p>In R, a t-test is just another function returning just another type of data, so it can also be a building block. The value it returns is a special type of vector called a “list”, but with a public face that presents itself nicely. This is a common pattern in R. Besides printing to the console nicely, this public face may alter the behaviour of generic functions such as <code>plot</code> and <code>summary</code>.</p>
<p>Similarly a data frame is a list of vectors that is able to present itself nicely.</p>
</div>
<div id="lists" class="section level3 extra unnumbered" number="">
<h3>Lists</h3>
<p>Lists are vectors that can hold anything as elements (even other lists!). It’s possible to create lists with the <code>list</code> function. This becomes especially useful once you get into the programming side of R. For example writing your own function that needs to return multiple values, it could do so in the form of a list.</p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="summarizing-data.html#cb216-1"></a>mylist &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">hello=</span><span class="kw">c</span>(<span class="st">&quot;Hello&quot;</span>,<span class="st">&quot;world&quot;</span>), <span class="dt">numbers=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>))</span>
<span id="cb216-2"><a href="summarizing-data.html#cb216-2"></a>mylist</span></code></pre></div>
<pre><code>## $hello
## [1] &quot;Hello&quot; &quot;world&quot;
## 
## $numbers
## [1] 1 2 3 4</code></pre>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="summarizing-data.html#cb218-1"></a><span class="kw">class</span>(mylist)</span></code></pre></div>
<pre><code>## [1] &quot;list&quot;</code></pre>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="summarizing-data.html#cb220-1"></a><span class="kw">typeof</span>(mylist)</span></code></pre></div>
<pre><code>## [1] &quot;list&quot;</code></pre>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="summarizing-data.html#cb222-1"></a><span class="kw">names</span>(mylist)</span></code></pre></div>
<pre><code>## [1] &quot;hello&quot;   &quot;numbers&quot;</code></pre>
<p>Accessing lists can be done by name with <code>$</code> or by position with <code>[[ ]]</code>.</p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="summarizing-data.html#cb224-1"></a>mylist<span class="op">$</span>hello</span></code></pre></div>
<pre><code>## [1] &quot;Hello&quot; &quot;world&quot;</code></pre>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="summarizing-data.html#cb226-1"></a>mylist[[<span class="dv">2</span>]]</span></code></pre></div>
<pre><code>## [1] 1 2 3 4</code></pre>
</div>
<div id="other-types-not-covered-here" class="section level3 extra unnumbered" number="">
<h3>Other types not covered here</h3>
<p>Matrices are another tabular data type. These come up when doing more mathematical tasks in R. They are also commonly used in bioinformatics, for example to represent RNA-Seq count data. A matrix, as compared to a data frame:</p>
<ul>
<li>contains only one type of data, usually numeric (rather than different types in different columns).</li>
<li>commonly has <code>rownames</code> as well as <code>colnames</code>. (Base R data frames can have <code>rownames</code> too, but it is easier to have any unique identifier as a normal column instead.)</li>
<li>has individual cells as the unit of observation (rather than rows).</li>
</ul>
<p>Matrices can be created using <code>as.matrix</code> from a data frame, <code>matrix</code> from a single vector, or using <code>rbind</code> or <code>cbind</code> with several vectors.</p>
<p>You may also encounter “S4 objects”, especially if you use <a href="http://bioconductor.org/">Bioconductor</a> packages. The syntax for using these is different again, and uses <code>@</code> to access elements.</p>
</div>
<div id="programming" class="section level3 extra unnumbered" number="">
<h3>Programming</h3>
<p>Once you have a useful data analysis, you may want to do it again with different data. You may have some task that needs to be done many times over. This is where programming comes in:</p>
<ul>
<li>Writing your own <a href="http://r4ds.had.co.nz/functions.html">functions</a>.</li>
<li><a href="http://r4ds.had.co.nz/iteration.html">For-loops</a> to do things multiple times.</li>
<li><a href="http://r4ds.had.co.nz/functions.html#conditional-execution">If-statements</a> to make decisions.</li>
</ul>
<p>The <a href="http://r4ds.had.co.nz/">“R for Data Science” book</a> is an excellent source to learn more. Monash Data Fluency <a href="https://monashdatafluency.github.io/r-progtidy/">“Programming and Tidy data analysis in R” course</a> also covers this.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="proteomics-data-viz.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="r-markdown.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["r-intro.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
