# Proteomics Data Viz

## Visualizing Proteomics data with ggplot2


```{r, include = FALSE}
knitr::opts_chunk$set(fig.width=6, fig.height=3, fig.align="center")
```

Last session we worked on Cross-linking Mass Spectrometry Data. The data consisted of interaction between 300 yeast nuclear proteins. 
We also learned about concepts behind `Grammer of Graphics` and plotting using `ggplot2`. 

In this tutorial, we will be working on the data from Yeast Nuclear Protein interaction study using Cross-linking Mass Spectrometry. 

```{r message=F, warning=F}
library(tidyverse)
```

We continue with the Cross-linking Proteomics dataset from Cytoscape tutorial.
The data was in excel, therefore we first converted it into `Comma separated file (csv)`  format.

### Optional: Reading excel file into R {.extra}
If you want to load the excel files files directly to `R` then you can use another library `readxl`. 

```{r message=F, warning=F}
library(readxl)
nuclear_xl_ms_excel <- readxl::read_excel("r-intro-files/Nuclear_XL_MS.xlsx")
head(nuclear_xl_ms_excel)
```


## Exploring the data
```{r message=F, warning=F}
library(readxl)
nuclear_xl_ms <- read_csv("r-intro-files/Nuclear_XL_MS.csv")
head(nuclear_xl_ms)
```

Now lets examine the dataset with two base R functions `str` and `summary`

```{r message=F, warning=F}
str(nuclear_xl_ms)
summary(nuclear_xl_ms)
```
While the data in `PPINovelty` and `PPIEvidenceInfoGroup` are characters/strings, they can also be thought of as categorical. 

We will now look into the unique values for the PPINovelty and PPIEvidenceInfoGroup columns

```{r message=F, warning=F}

nuclear_xl_ms$PPINovelty %>% unique()
```
```{r message=F, warning=F}
nuclear_xl_ms$PPIEvidenceInfoGroup %>% unique()
```


R has a class for categorical data known as factors. We can convert these columns to factors and provide an order to those categories (levels). By default, R will order the levels of a factor alphabetically but we can override this behaviour by defining the level order.
Here we will order the Evidence based on Strength of evidence for the interaction


```{r message=F, warning=F}
nuclear_xl_ms <- nuclear_xl_ms %>% mutate(
  PPINovelty = factor(PPINovelty),
  PPIEvidenceInfoGroup = factor(PPIEvidenceInfoGroup, levels = c("Structure","APID", "STRING", "Genetic", "Unexplained"))
)
nuclear_xl_ms %>% str()
```

## Plotting interactions types
Firstly we will plot the number of Known and novel interactions using with `geom_bar`

```{r message=F, warning=F}
ggplot(nuclear_xl_ms, aes(x = PPINovelty)) + 
geom_bar() + 
        xlab("Novelty") +
  theme_bw()
```

Now, we will rotate the bars to **Y-Axis** using `coord_flip()"`

```{r message=F, warning=F}
ggplot(nuclear_xl_ms, aes(x = PPINovelty)) + 
 geom_bar(position = "dodge") + 
        xlab("Novelty") +
        coord_flip() +
  theme_bw()
```

Next, step would be to create a stacked bar chart by adding **PPIEvidenceInfoGroup** data on top of each bar

```{r message=F, warning=F}
ggplot(nuclear_xl_ms, aes(PPINovelty)) + 
  geom_bar(aes(fill=PPIEvidenceInfoGroup)) + 
        xlab("Novelty") +
        coord_flip() +
  theme_bw()
```


## Individual Proteins

First we will calculate how many times a protein appeared in **NameProtein1** column using `table` function and then sorting by descending order.
Next we will use `head()` function to print first five proteins with most observation. 

```{r message=F, warning=F}
table(nuclear_xl_ms$NameProtein1) %>% sort(.,decreasing = TRUE) %>% head()
```

We could do same thing for **NameProtein2** column as well. 

```{r message=F, warning=F}
table(nuclear_xl_ms$NameProtein2)  %>% sort(.,decreasing = TRUE) %>% head()
```

Now we store the rows containing H2B1 and EF1A proteins in **NameProtein1** column in a datafame.

```{r message=F, warning=F}
two_protein_df<- nuclear_xl_ms %>% filter(NameProtein1 %in% c("H2B1","EF1A"))

ggplot(two_protein_df, aes(x=PPINovelty, y=NumberUniqueLysLysContacts)) +
  geom_col(aes(fill=PPIEvidenceInfoGroup)) +
  labs(x= "Novelty",
  y= "Number of Contacts") +
  facet_wrap(~NameProtein1)+
  theme_bw()
```

## Volcano Plot {.extra -}

In this section, we will see how to plot a volcano plot for a quantitative proteomics dataset. 
This dataset is derived from label-free quantitative proteomics experiment investigating differences in protein profiles between Benign and 
Malignant Prostate cancers. 

The details can be found on [LFQ-Analyst](https://bioinformatics.erc.monash.edu/apps/LFQ-Analyst/) under the **Demo** tab.

There are 20 samples in total with `n=10` in each group. 

A moderated _t-test_ was performed to find differentially expressed proteins in the dataset. Each row represents a protein along with log fold change and p-values.

In this tutorial we will visualise the results in the form of **Volcano Plot**.

Firstly, we will load the data. 

```{r message=F, warning=F}
lfq_data<-read_csv("r-intro-files/LFQ-Analyst_results.csv")
nrow(lfq_data)
ncol(lfq_data)
```
The data has quantitative information about 2389 proteins and has 10 columns.

Now lets see the column names.

```{r message=F, warning=F}
colnames(lfq_data)
```

For plotting the volcano plot, we need to focus on th three columns 

1. Benign_vs_Malignant_log2 fold change
2. Benign_vs_Malignant_p.adj
3. significant

Next we will on the fly convert FDR values to `-log10` and plot it against log2 fold change on the **X-axis**

```{r message=F, warning=F}
volcano_plot <- ggplot(lfq_data, 
                       aes(x = `Benign_vs_Malignant_log2 fold change`, 
                           y = -log10(`Benign_vs_Malignant_p.adj`))) + 
  geom_point(color="grey") + 
  labs(x= "Log 2 fold change",
       y= "-log 10 FDR") + 
  theme_bw()
volcano_plot

volcano_plot + geom_point(data= filter(lfq_data, significant=="TRUE"), color="black")
```



